src/System.Net.HttpListener/tests/HttpRequestStreamTests.cs

```diff
@@ -505,13 +505,10 @@ public void Dispose()
                 byte[] buffer = new byte[expected.Length];
                 await Assert.ThrowsAsync<HttpListenerException>(() => context.Request.InputStream.ReadAsync(buffer, 0, buffer.Length));
                 await Assert.ThrowsAsync<HttpListenerException>(() => context.Request.InputStream.ReadAsync(buffer, 0, buffer.Length));
-
-                // Closing a response from a closed client if no writing has failed should fail.
-                Assert.Throws<HttpListenerException>(() => context.Response.Close());
```

Comment:
> Because it seemed like weird behavior. The client has sent a malformed request; once I get the exception on reading the request, what else am I supposed to do except Close? I would understand if Write threw, but Close seems weird.

I'd argue the point of tests like this is to demonstrate the weird behaviour! I'm perhaps unreasonably against deleting tests ever without good reason.

>If this is Windows behavior, then maybe we should consider whether that should change.

Unlikely that's going to happen - HttpListener in netcoreapp is a legacy API and breaking changes aren't likely to be accepted. From @karelz in #19437

> I would suggest to close it, unless:

> We see customers hitting it hard (which is unlikely as it was not reported even on .NET Framework AFAIK)
> Fix is simple and someone submits it
> Otherwise I would not invest in legacy APIs to this level ... thoughts?